src/applications/vaos/containers/VAFacilityPage.jsx

```diff
@@ -65,9 +66,26 @@ const uiSchema = {
     },
   },
   vaFacilityMessage: {
-    'ui:field': ({ formContext }) => (
-      <NoValidVAFacilities systemId={formContext.vaSystem} />
-    ),
+    'ui:field': ({ formContext }) => {
+      const {
+        vaSystem,
+        typeOfCare,
+        facilityDetailsStatus,
+        systemDetails,
+      } = formContext;
+
+      if (facilityDetailsStatus === FETCH_STATUS.loading) {
```

Comment:
I guess I don't have a strong opinion, but it feels a little weird to have this much code inline in the `ui:field` property. But, the way we have to create this message is weird overall. An alternative would be to make `NoValidFacilties` a connect component and pull in the data it needs from Redux itself, without passing everything through formContext.

Reply:
I cleaned this up a bit by moving the loading indicator logic into the <NoValidVaFacilities/> component, reducing the code in the `'ui:field` by quite a bit.